### PR TITLE
Feature/clean res1d params

### PR DESCRIPTION
**Pull request by ryan-kipawa**

### CODE REVIEW

File: mikeio1d/res1d.py [hunk 1]
```diff
@@ -107,19 +112,37 @@ class Res1D:
...
+        warn_deprecation("clear_queue_after_reading")
+        warn_deprecation("header_load", "Dynamic data is read lazily, so header_load is not needed.")
+        warn_deprecation("result_reader_type", "Use res1d.result_reader_type instead.")
+
+        lazy_load = kwargs.get("lazy_load", False)
+        col_name_delimiter = kwargs.get("col_name_delimiter", NAME_DELIMITER)
+        put_chainage_in_col_name = kwargs.get("put_chainage_in_col_name", True)
+        clear_queue_after_reading = kwargs.get("clear_queue_after_reading", True)
+        header_load = kwargs.get("header_load", False)
+        result_reader_type = kwargs.get("result_reader_type", ResultReaderType.COPIER)
```
Comment (gedaskir) [inline mikeio1d/res1d.py:141]: I thought we agreed that this one we do not deprecate :-).

File: mikeio1d/res1d.py [hunk 2]
```diff
@@ -107,19 +112,37 @@ class Res1D:
...
+        warn_deprecation("col_name_delimiter")
+        warn_deprecation("put_chainage_in_col_name")
+        warn_deprecation("clear_queue_after_reading")
+        warn_deprecation("header_load", "Dynamic data is read lazily, so header_load is not needed.")
+        warn_deprecation("result_reader_type", "Use res1d.result_reader_type instead.")
+
+        lazy_load = kwargs.get("lazy_load", False)
+        col_name_delimiter = kwargs.get("col_name_delimiter", NAME_DELIMITER)
+        put_chainage_in_col_name = kwargs.get("put_chainage_in_col_name", True)
+        clear_queue_after_reading = kwargs.get("clear_queue_after_reading", True)
```
Comment (gedaskir) [inline mikeio1d/res1d.py:139]: I used this API with add()...

File: mikeio1d/res1d.py [hunk 3]
```diff
@@ -157,25 +156,11 @@ def warn_deprecation(name: str, hint: str = ""):
             time=time,
         )
 
-        self.result_network = ResultNetwork(self)
-        self.network = self.result_network
+        self._network = ResultNetwork(self)
```
Comment (gedaskir) [inline mikeio1d/res1d.py:159]: Do we really want to underscore this one? It is not private to Res1D class, but internal to mikeio1d.

File: mikeio1d/res1d.py [hunk 4]
```diff
@@ -488,6 +488,11 @@ def start_time(self) -> datetime:
     def end_time(self) -> datetime:
         """End time of the result file."""
         return from_dotnet_datetime(self.data.EndTime)
+    
+    @property
+    def n_time_steps(self) -> int:
```
Comment (gedaskir) [inline mikeio1d/res1d.py:493]: I suggest to consider less cryptic naming: number_of_time_steps vs. n_time_steps :-).

**Review by gedaskir — APPROVED** [review]
Just minor comments :-).

File: mikeio1d/res1d.py (hunk 2 above)
Comment (ryan-kipawa) [reply mikeio1d/res1d.py:139]: It's still accessible via Res1D.clear_queue_after_reader, just not as an instantiation parameter

File: mikeio1d/res1d.py (hunk 1 above)
Comment (ryan-kipawa) [reply mikeio1d/res1d.py:141]: I can't recall 🙈 It's still possible to change the reader via Res1D.reader = ResultReaderCreator.create(...). I'm in favor of as clean as possible of a parameter set for the public scripting API. Unless you're using it very often in the constructor, then I'm inclined to remove it. What do you think? @gedaskir

File: mikeio1d/res1d.py (hunk 3 above)
Comment (ryan-kipawa) [reply mikeio1d/res1d.py:159]: Good point. I put it as a read only property instead. But I guess it's also fine to leave it as something that someone could hack with if they want. Subclassing and creating their own super network object 😆

File: mikeio1d/res1d.py (hunk 4 above)
Comment (ryan-kipawa) [reply mikeio1d/res1d.py:493]: Good point, I'll change that

**Comment by ryan-kipawa** [comment]
Thanks for the comments Gedi. Made some changes based on them. I'll merge as is now, but we can revisit adding back some parameters to Res1D if they're truly important

**Pull request merged**